### ISSUE TITLE
Add type hints

### ISSUE THREAD

**Issue by karolzlot** [elapouya/python-docx-template#441]
## Is your feature request related to a problem? Please describe.
Currently source code lacks type hints. Adding them can improve convenience of using library and also make library maintenance easier (find easier bugs / edge cases etc.).

## Describe the solution you'd like
Add type hints to source code.

## Describe alternatives you've considered
n/d

## Additional context
It will help to some extent with such issues: https://github.com/elapouya/python-docx-template/issues/416

----

I can create PR with some type hints added, but I would like first to know if it is likely to be accepted (from my experience not every maintainer likes adding type hints to source code due to being used to code without type hints).

**Comment by elapouya** [comment]
Yes it is a good idea (package was created with python2.x, this was not available then). I will be very happy to have a PR from you :)
